scheduler/reconcile.go

```diff
@@ -555,9 +554,12 @@ func (a *allocReconciler) computeGroup(group string, all allocSet) bool {
 		// A previous group may have made the deployment already
 		if a.deployment == nil {
 			a.deployment = structs.NewDeployment(a.job)
-			// only the first region of a multiregion job starts in the
-			// running state
-			if a.job.IsMultiregion() && a.job.Region != a.job.Multiregion.Regions[0].Name {
+			// in a multiregion job, if max_parallel is set, only the first
+			// region starts in the running state
+			if a.job.IsMultiregion() &&
+				a.job.Multiregion.Strategy != nil &&
+				a.job.Multiregion.Strategy.MaxParallel != 0 &&
```

Comment:
I missed some of the changes of PRs - do we also want to check MaxParallel against current region index (e.g. second region with MaxParallel=2); is that handled elsewhere?

Reply:
Oh that's a good catch. That's the same sort of thing -- we'd safely have _at most_ `max_parallel` going, but the operator probably expects us to start with `max_parallel`. Will fix.

Reply:
The logic here is getting gnarly though so I'm going to pull it out into a function.

Reply:
I've pulled this logic out to a function on `Job` in f10cc93